controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -157,4 +208,28 @@ protected void shutDown() throws Exception {
             LoggerHelpers.traceLeave(log, objectId, "shutDown", traceId);
         }
     }
+
+    @VisibleForTesting
+    static class PendingSweepQueue {
+        private final ConcurrentLinkedQueue<Supplier<CompletableFuture<Void>>> workQueue = new ConcurrentLinkedQueue<>();
+        private final AtomicBoolean isSealed = new AtomicBoolean(false);
+
+        @Synchronized
+        @VisibleForTesting
+        boolean enqueue(Supplier<CompletableFuture<Void>> futureSupplier) {
```

Comment:
I am wondering whether this pending queue is really necessary, because we need not store host lost events before eventProcessors or txnSweeper are ready. Once they get ready they would evaluate the set of failed hosts and sweep their event readers and txns, respectively.

If we sweep tasks for failed hosts before waiting for eventProcessors and txnSweepers to get ready, then that would be a simple fix for this issue, without the need for additional queue.